.github/workflows/build_and_test.yml

```diff
@@ -111,8 +111,18 @@ jobs:
       - name: Get postgres headers
         run: make postgres-headers -j$(nproc)
 
-      - name: Run cargo clippy
-        run: ./run_clippy.sh
+      - uses: taiki-e/install-action@cargo-hack
+
+      # cargo hack runs the given cargo subcommand (clippy in this case) for all feature combinations.
+      # This will catch compiler & clippy warnings in all feature combinations.
+      # TODO: use cargo hack for build and test as well, but, that's quite expensive.
+      # NB: keep clippy args in sync with ./run_clippy.sh
+      - run: |
+          echo "CLIPPY_COMMON_ARGS=--locked --workspace --all-targets -- -A unknown_lints -D warning" >> $GITHUB_ENV
```

Comment:
This is annoying how clippy cannot have a config file (last checked when I modified the run_clippy.sh).

But, we don't change these much so perhaps it would be good.

Reply:
I suppose we could have our own file that contains these common args, then both ./run_clippy.sh and this job could read that file.

Reply:
The common file idea is looking good.